torch/_dynamo/source.py

```diff
@@ -322,9 +322,9 @@ def name(self):
         # 3) index is a slice - example 1:4
         # 4) index is a constant - example string, integer
         if isinstance(self.index, Source):
-            if not isinstance(self.index, ConstDictKeySource):
+            if not isinstance(self.index, (ConstDictKeySource, ClassSource)):
```

Comment:
I was hoping that we could just use the ConstDictKeySource to access the key. For a code like this (this passes on main)

~~~
import torch

class A:
    value = 5

d = {
    A : 2
}

@torch.compile(backend="eager")
def fn(x):
    return x * d[A]

fn(torch.randn(4))
~~~

This is the guard that we get ` ___check_obj_id(list(G['d'].keys())[0], 8593344)`

I am thinking can we do the same thing here with SUPPORTED_NODES. Basically when we do a get item on a ConstDictVariable (I am hoping SUPPORTED_NODES point to a ConstDictVariable), we can insert a ConstDictKeySource.